### DISCUSSION TITLE
Ground rules for colocated repos?

### DISCUSSION THREAD

**Discussion by sparkybrit** [jj-vcs/jj#5211] [Q&A]
`jj new` often seems to cause a “detached head” in a colocated repo.

What are the ground rules (e.g. always bookmarking branches etc) to ensure no "detached heads"?

**Comment by bnjmnt4n** [comment]
There is currently no way to avoid the detached HEAD state in colocated repositories.

There is a WIP PR to do so when the `experimental-advance-branches` option is enabled: https://github.com/jj-vcs/jj/pull/3405.

**Comment by PhilipMetzger** [comment]
> What are the ground rules (e.g. always bookmarking branches etc) to ensure no "detached heads"?

No, Jujutsu has special keep refs which keeps detached refs alive, see https://github.com/jj-vcs/jj/discussions/4848#discussioncomment-11242238

The detached head is something to get used to, but you won't notice it if you keep on using `jj`.
